tests/test_text_cleaning.py

```diff
@@ -0,0 +1,413 @@
+        self.assertIn('Latin', scripts)
+        self.assertIn('Cyrillic', scripts)
+        
+        # Numbers and punctuation
+        scripts = self.filter.analyze_word_scripts("123!@#")
+        self.assertEqual(len(scripts), 0)  # Numbers/punctuation don't have scripts
+    
+    def test_remove_non_latin_words_basic(self):
+        """Test basic non-Latin word removal."""
+        text = "Hello world привіт світ"
+        result = self.filter.remove_non_latin_words(text)
+        self.assertEqual(result, "Hello world")
+        
+        text = "Test 123 тест 456"
+        result = self.filter.remove_non_latin_words(text)
+        self.assertEqual(result, "Test 123 456")
+    
+    def test_remove_non_latin_words_preserve_mixed(self):
+        """Test non-Latin word removal with mixed word preservation."""
+        text = "Hello world привітworld"
+        result = self.filter.remove_non_latin_words(text, preserve_mixed=True)
+        self.assertEqual(result, "Hello world привітworld")  # Mixed word preserved
+        
+        text = "Hello world привіт"
+        result = self.filter.remove_non_latin_words(text, preserve_mixed=True)
+        self.assertEqual(result, "Hello world")  # Pure non-Latin word removed
+    
+    def test_clean_text_aggressive(self):
+        """Test aggressive text cleaning."""
+        text = "Hello world привіт світ 123!@#"
+        result = self.filter.clean_text_aggressive(text)
+        self.assertEqual(result, "Hello world 123!@#")  # Non-Latin removed
+    
+    def test_clean_text_aggressive_edge_cases(self):
+        """Test aggressive cleaning with edge cases."""
+        # Empty text
+        self.assertEqual(self.filter.clean_text_aggressive(""), "")
+        self.assertEqual(self.filter.clean_text_aggressive(None), None)
+        
+        # Only non-Latin
+        self.assertEqual(self.filter.clean_text_aggressive("привіт світ"), "")
+        
+        # Mixed with special characters
+        text = "Hello привіт!@#$%^&*()"
+        result = self.filter.clean_text_aggressive(text)
+        self.assertIn("Hello", result)
+        self.assertIn("!@#$%^&*()", result)
+        self.assertNotIn("привіт", result)
+
+
+class TestAttractionsParserCleaning(unittest.TestCase):
+    """Test the text cleaning methods in AttractionsParser."""
+    
+    def setUp(self):
+        """Set up test fixtures."""
+        self.parser = AttractionsParser()
+    
+    def test_remove_see_also_section(self):
+        """Test removal of 'See also' sections."""
+        text = """
+        Some content here.
+        
+        == See also ==
+        * Link 1
+        * Link 2
+        
+        More content.
+        """
+        result = self.parser._remove_see_also_section(text)
+        self.assertNotIn("== See also ==", result)
+        self.assertNotIn("Link 1", result)
+        self.assertIn("Some content here", result)
+        self.assertIn("More content", result)
```

Comment:
add cases where "See also" is embedded in other text or is formatted differently (e.g., extra whitespace, tabs)